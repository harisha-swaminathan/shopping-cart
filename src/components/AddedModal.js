import React from 'react';
import Modal from 'react-modal';

class AddedModal extends React.Component {
  state={
    isOpen:true
  };
  ClearSelectedOption=()=>{
    this.setState({
      isOpen:false
    })
  };
  render(){
    return(
      <Modal
      isOpen={!!this.state.isOpen}
      onRequestClose={this.ClearSelectedOption}
      contentLabel="Item added to cart"
      shouldCloseOnOverlayClick={true}
      closeTimeoutMS={200}
      className="modal"
    >
      <h3>Item added to cart!</h3>
         
         <button className="button-color" onClick={this.ClearSelectedOption}>Okay</button>
    </Modal>
    );
  }
}


export default AddedModal;
