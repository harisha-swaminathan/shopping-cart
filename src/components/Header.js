import React from 'react';
import { Link } from 'react-router-dom';
import {connect} from 'react-redux';
import {startLogout} from '../actions/auth';
export const Header = ({startLogout}) => (
  <header className="header-color">
  <div className="container">
  <div className="header-color__content">
    <Link className="header-color__title" to="/dashboard"><h1> shopping </h1></Link>
    <button className="button-color button-merge" onClick={startLogout}>Logout </button>
    </div>
    </div>
  </header>
);

const mapDispatchToProps=(dispatch)=>({
  startLogout:()=> dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);
