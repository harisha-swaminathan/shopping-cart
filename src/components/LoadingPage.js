import React from 'react';

const LoadingPage=()=>(
    <div className="loader">
        <img className="loader__image" src="/loader.gif"></img>
    </div>

);

export default LoadingPage;