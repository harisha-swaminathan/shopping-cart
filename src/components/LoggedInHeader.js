import React from 'react';
import { Link } from 'react-router-dom';
import {connect} from 'react-redux';
import {startLogout} from '../actions/auth';
export const LoggedInHeader = ({startLogout}) => (
  <header className="header-color">
  <div className="container">
  <div className="header-color__content">
  <Link to="/dashboard"><img alt="book" src="logo.png" className="logo"/></Link>
  <h2 className="header-color__title" > Your one stop shop for front end development </h2>
  <div className="header-color__right">
  <Link to="/cart"><img alt="cart" src="cart.png" className="logo_cart"/></Link>
  <button className="button-color" onClick={startLogout}>Logout </button>
  </div>
    </div>
    </div>
  </header>
);

const mapDispatchToProps=(dispatch)=>({
    startLogout:()=> dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(LoggedInHeader);