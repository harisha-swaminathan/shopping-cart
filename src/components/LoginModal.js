import React from 'react';
import Modal from 'react-modal';

class LoginModal extends React.Component {

  state={
    isOpen:true
  };
  ClearSelectedOption=()=>{
    this.setState((prev)=>({
      isOpen:!prev
    }));
  };
  componentWillReceiveProps() {
    this.setState({
      isOpen : true
    });
  }
  render(){
    console.log(this.props);
    return(
      <Modal
      isOpen={!!this.state.isOpen}
      onRequestClose={this.ClearSelectedOption}
      contentLabel="Please Login "
      shouldCloseOnOverlayClick={true}
      closeTimeoutMS={200}
      className="modal"
    >
      <h3>Please Login to add item to cart! </h3>
         
         <button className="button-color" onClick={this.ClearSelectedOption}> Okay</button>
    </Modal>
    );
  }
}


export default LoginModal;
