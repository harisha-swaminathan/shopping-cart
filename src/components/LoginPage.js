import React from 'react';
import {connect} from 'react-redux';
import {startLogin} from '../actions/auth';

export const LoginPage=(props)=>(
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title"> shopping cart</h1>
     <button className="button-color" onClick={props.startLogin}>Login with Google</button>
    </div>
        
      </div>
);

const mapDispatchToProps =(dispatch)=>({
  startLogin: ()=>dispatch(startLogin())
});

export default connect(undefined,mapDispatchToProps)(LoginPage);