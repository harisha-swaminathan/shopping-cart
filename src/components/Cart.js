import React from 'react';
import { connect } from 'react-redux';
import ProductsInCart from './ProductsInCart';
import TotalSummary from './TotalSummary';
import Header from './Header';
export const Cart = (props) => (
   <div >
     <Header/>
    <div className="container"> 
    <h1>Summary</h1>
    <div className="list-header">
    <div className="visible-desktop">Product</div>
    <div className="visible-desktop">Quantity</div>
    <div className="visible-desktop">Remove</div>
    <div className="visible-desktop">Price</div>
    </div>
    
  <div className="list-bottom">
    {
      props.products.length === 0 ? (
        <div> 
          <span className=" list-item list-item--empty">
           No Products in cart
          </span>
        </div>
      ) : (
          props.products.map((product) => {
            return <ProductsInCart key={product.pid} {...product} />;
          })
        )
    }
    <div className="summary-heading">
    <h3 >Total</h3>
    <TotalSummary/>
    <button  className="button-color__remove">Checkout</button>
    </div>
    </div>
  </div>
   </div>
  
);


const mapStateToProps = (state) => {
  return {
    products: state.products
  };
};

export default connect(mapStateToProps)(Cart);
