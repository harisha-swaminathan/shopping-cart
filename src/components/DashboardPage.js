import React from 'react';
import LoggedInHeader from './LoggedInHeader'
import ProductList from './ProductList';
import {storeProducts} from '../fixtures/products';
const DashboardPage = () => (
  <div>
    <LoggedInHeader/>
      <h1>shows same product page with different Header</h1>
      <ProductList products={storeProducts}/>
  </div>
);

export default DashboardPage;

