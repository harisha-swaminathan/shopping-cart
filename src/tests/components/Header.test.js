import React from 'react';
import { shallow } from 'enzyme';
import {Header} from '../../components/Header';

test('should render Header correctly', () => {
  const wrapper = shallow(<Header startLogout={()=>{}} />);
  expect(wrapper).toMatchSnapshot();
});

const startLogout =jest.fn();
const wrapper = shallow(<Header startLogout={startLogout} />);
test('call start log in on button click',()=>{
  wrapper.find('button').simulate('click');
  expect(startLogout).toBeCalled();
});