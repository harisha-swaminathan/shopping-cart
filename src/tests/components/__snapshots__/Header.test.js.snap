// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header
  className="header-color"
>
  <div
    className="container"
  >
    <div
      className="header-color__content"
    >
      <Link
        className="header-color__title"
        replace={false}
        to="/dashboard"
      >
        <h1>
           shopping 
        </h1>
      </Link>
      <button
        className="button-color button-merge"
        onClick={[Function]}
      >
        Logout 
      </button>
    </div>
  </div>
</header>
`;
